python/Scripts/generateshader.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+'''
+Utility to generate the shader for materials found in a MaterialX document. One file will be generated
+for each material / shader found. The currently supported target languages are GLSL, OSL, MDL and ESSL.
+'''
+
+import sys, os, argparse, subprocess
+import MaterialX as mx
+import MaterialX.PyMaterialXGenShader as mx_gen_shader
+import MaterialX.PyMaterialXGenGlsl as mx_gen_glsl
+import MaterialX.PyMaterialXGenOsl as mx_gen_osl
+import MaterialX.PyMaterialXGenMdl as mx_gen_mdl
+import MaterialX.PyMaterialXGenEssl as mx_gen_essl
+
+
+def validateCode(sourceCodeFile, codevalidator, codevalidatorArgs):
+    if codevalidator and os.path.isfile(codevalidator):
+        cmd = codevalidator + ' ' + sourceCodeFile 
+        if codevalidatorArgs:
+            cmd + ' ' + codevalidatorArgs
+        print('----- Run: '+ cmd)
+        try:
+            output = subprocess.check_output(cmd, stderr=subprocess.STDOUT)
+            result = output.decode(encoding='utf-8')
+            print('----- Validation success: ' + result)
+        except subprocess.CalledProcessError as out:                                                                                                   
+            print('----- Validation failed: ', out.returncode)
+            print('    Result: ' + out.output.decode(encoding='utf-8'))
+    else:
+        print('--- No validation performed')
+
+def main():
+    parser = argparse.ArgumentParser(description='Generate shader code for each material / shader in a document.')
+    parser.add_argument('--path', dest='paths', action='append', nargs='+', help='An additional absolute search path location (e.g. "/projects/MaterialX")')
+    parser.add_argument('--library', dest='libraries', action='append', nargs='+', help='An additional relative path to a custom data library folder (e.g. "libraries/custom")')
+    parser.add_argument('--target', dest='target', default='glsl', help='Target shader generator to use (e.g. "genglsl"). Default is genglsl.')
+    parser.add_argument('--outputPath', dest='outputPath', help='File path to output shaders to.')
+    parser.add_argument('--validator', dest='validator', nargs='?', const=' ', type=str, help='Name of executable to perform source code validation.')
```

Comment:
@jstone-lucasfilm , I think this should be generic enough for this script to pass in
an argument for the validator + validator args. I separated the 2 as the script checks for
existence of the validator then builds the command to run.